runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/retries/impl/StandardRetryTokenBucket.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package aws.smithy.kotlin.runtime.retries.impl
+
+import aws.smithy.kotlin.runtime.retries.RetryErrorType
+import aws.smithy.kotlin.runtime.retries.RetryToken
+import aws.smithy.kotlin.runtime.retries.RetryTokenBucket
+import aws.smithy.kotlin.runtime.time.Clock
+import aws.smithy.kotlin.runtime.time.epochMilliseconds
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlin.math.ceil
+import kotlin.math.floor
+import kotlin.math.min
+
+/**
+ * The standard implementation of a [RetryTokenBucket].
+ * @param options The configuration to use for this bucket.
+ */
+class StandardRetryTokenBucket(
```

Comment:
question

What makes this `Standard`?

Reply:
question

This only supports `standard` not `adaptive` behavior correct?

Reply:
> What makes this `Standard`?

This is standard in that it's recommended and the only implementation provided by **smithy-kotlin**.

> This only supports `standard` not `adaptive` behavior correct?

For now, yes that's correct.